brasilio_auth/scripts/migrate_wrong_usernames.py

```diff
@@ -38,23 +39,29 @@ def possible_usernames(username, email):
     email_parts = email.split("@")
     possible_2 = email_parts[0]
     possible_3 = email_parts[0] + "_" + email_parts[1].split(".")[0]
-    return (username, possible_2, possible_3)
-
-
-def migrate_usernames():
-    # with open("/data/fixed-usernames.csv", mode="w") as fobj:
-    # writer = csv.DictWriter(fobj, fieldnames=["old_username", "new_username", "email"])
-    for user in User.objects.filter(username__contains="@"):
-        possible = possible_usernames(user.username, user.email)
-        changed = False
-        for username in possible:
-            if not User.objects.filter(username=username).exists():
-                # writer.writerow(...)
-                changed = True
-                # TODO: save user.email in CSV
-                # TODO: create script to send email
-                #user.username = new
-                #user.save()
-                pass
-        if not changed:
-            print(f"Conflict in: {user.username} / {user.email} (would be: {new} but already exists)")
+    possible_with_suffix = [f"{username}_{i}" for i in range(1, n_suffix)]
+    return (username, possible_2, possible_3, *possible_with_suffix)
+
+
+def migrate_usernames(filepath=None):
+    filepath = filepath or "/data/fixed-usernames.csv"
+    with open(filepath, mode="w") as fobj:
+        writer = csv.DictWriter(
+            fobj, fieldnames=["old_username", "new_username", "email"]
+        )
+        writer.writeheader()
+        for user in User.objects.filter(username__contains="@"):
+            possible = possible_usernames(user.username, user.email)
+            changed = False
+            for username in possible:
+                if not User.objects.filter(username=username).exists() and not changed:
```

Comment:
Aqui não precisaria do `not changed` se tivesse um `break` no fim do bloco do `if`, depois do `user.save()`.

Reply:
solved by 8d6bddccd0f6b9fe32